galaxy_importer/ansible_test/runner.py

```diff
@@ -0,0 +1,58 @@
+# (c) 2012-2020, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+from galaxy_importer import config
+import logging
+
+
+default_logger = logging.getLogger(__name__)
+
+
+class AnsibleTestRunner(object):
+    """Runs ansible-test on collection, dependent on config."""
+
+    def __init__(self, logger=None):
+        self.log = logger or default_logger
+        self.cfg = config.Config()
+
+    def run(self):
+        if not self.cfg.run_ansible_test:
+            return
+
+        if not self.cfg.infra_pulp:
+            self._run_ansible_test_local()
+            return
+
+        image = self._build_img_with_artifact()
+
+        if self.cfg.infra_osd:
+            self._run_image_openshift_job(image)
+        else:
+            self._run_image_local(image)
```

Comment:
Since this class currently does 3 or 4 things:
- decide which way run ansible test
- run ansible-test local
- build a container image
- run container image in openshift
- run container image locally

May be easier to:
- split the config based decision to it's own method (module level, or even elsewhere )
- define the AnsibleTestRunner "public" API  (ie, just the init and run())
- provides implementations for LocalAnsibleTestRunner(), OpenshiftJobTestRunner(), LocalImageTestRunner()

That may make it easier to make a plugin or entrypoint at some point.
Would also make the impl classes more focused.
Would let the impl classes be in separate modules with just the python modules they need (ie, so only ansible_test.runner.openshift_runner is the only modules that needs to import openshift modules)

Reply:
I think splitting it up like this makes sense